Source/CarthageKit/Project.swift

```diff
@@ -960,6 +960,22 @@ public final class Project {
 			}
 	}
 
+	public func generateSwiftPackageManagerXcodeProjectIfAvailable(forDependencyAt dependencyPath: String) -> SignalProducer<(), NoError> {
+		let files = (try? FileManager.default.contentsOfDirectory(atPath: dependencyPath)) ?? []
+
+		let hasXcodeProjectFile = files.lazy.filter { $0.hasSuffix(".xcodeproj") || $0.hasSuffix(".xcworkspace") }.first != nil
+		guard !hasXcodeProjectFile else { return .init(value: ()) }
+
+		let hasPackageFile = files.lazy.filter { $0 == "Package.swift" }.first != nil
+		guard hasPackageFile else { return .init(value: ()) }
+
+		let task = Task("/usr/bin/swift", arguments: [ "package", "generate-xcodeproj" ], workingDirectoryPath: dependencyPath)
```

Comment:
Generating an Xcode project to the dependency directory makes it dirty so that should be avoided. You should generate a project to a temporary directory.